src/main/java/io/opentracing/contrib/global/GlobalSpanManager.java

```diff
@@ -0,0 +1,87 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.Span;
+
+import java.io.Closeable;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class manages the globally 'active' {@link Span} in the current Thread.
+ *
+ * @author Sjoerd Talsma
+ */
+public final class GlobalSpanManager {
+
+    private static final Logger LOGGER = Logger.getLogger(GlobalSpanManager.class.getName());
+
+    private static final ThreadLocal<ManagedSpan> ACTIVE_SPAN = new InheritableThreadLocal<ManagedSpan>();
```

Comment:
@adriancole Thank you for those two links.
I agree it may be safest to be explicit when crossing thread-boundaries; I propose to change the `InheritableThreadLocal` to `ThreadLocal`. By the way; I think the Log4j discussion fundamentally forgets to 'close' their contexts; each span obtained from the Global tracer should -when finished- restore the previous active span (i.e. 'pop' itself in stack terms).

I absolutely like the `ContextStorageOverride` approach in the grpc link and will incorporate a similar mechanism in my `GlobalSpanManager`.